src/actions/sources/newSources.js

```diff
@@ -79,15 +79,19 @@ function loadSourceMap(sourceId: SourceId) {
 
     if (!urls) {
       // If this source doesn't have a sourcemap, enable it for pretty printing
-      dispatch(
+      return dispatch(
         ({
           type: "UPDATE_SOURCE",
           source: { ...source, sourceMapURL: "" }
         }: Action)
       );
-      return;
     }
-
+    dispatch(
+      ({
+        type: "UPDATE_SOURCE",
+        source: { ...source, url: `${source.url}[sm]` }
```

Comment:
i think we only want to add [sm] *if* there is a generated source with the same URL